scripts/utils/trading_strategy_helpers.js

```diff
@@ -275,14 +329,11 @@ const checkSufficiencyOfBalance = async function(token, owner, amount) {
  * @return {Transaction} Multisend transaction that has to be sent from the master address to either request
 withdrawal of or to withdraw the desired funds
 */
-const buildGenericFundMovement = async function(masterAddress, withdrawals, functionName, web3 = web3, artifacts = artifacts) {
-  BatchExchange.setProvider(web3.currentProvider)
-  BatchExchange.setNetwork(web3.network_id)
-  const exchange = await BatchExchange.deployed()
-  const masterTransactions = []
-
-  // it's not necessary to avoid overlapping withdraws, since the full amount is withdrawn for each entry
-  for (const withdrawal of withdrawals) {
+const buildGenericFundMovement = async function(masterAddress, withdrawals, functionName, web3 = web3, artifacts = artifacts) { // TODO: do we need artifacts here?
+  const exchange = await getExchange(web3)
+  
+  // it's not necessary to avoid overlapping withdraws, since the full amount is withdrawn for each entry TODO: is this still true?
+  const masterTransactionsPromises = withdrawals.map((withdrawal) => (async () => {
     // create transaction for the token
     let transactionData
     switch (functionName) {
```

Comment:
Does `encodeABI()` really return a promise?

Reply:
No, you are right.
Await is used everywhere in this project when `encodeABI` is invoked, I'll fix that in another PR since it requires some other changes (hopefully, no need to retrieve MultiSend and GnosisSafeMasterCopy from the network anymore, I'll think about it.).